lib/TPP/DefaultPipeline.cpp

```diff
@@ -46,6 +46,12 @@ llvm::cl::opt<bool>
                 llvm::cl::desc("Default pipeline - enable parallel execution"),
                 llvm::cl::init(false));
 
+// Control grid parallelism sizes.
+llvm::cl::opt<std::string>
+    parallelTaskGrid("parallel-task-grid",
+                     llvm::cl::desc("Grid-sizes for parallel tasks"),
+                     llvm::cl::init("2,8"));
```

Comment:
What would it be the right arguments for the OMP benchmarks we have?

Can you add them to the BF16 benchmarks to see if it improves from the current state?

Here's my calculations on speed up from baseline:
| threads | speedup |
| -- | -- |
| 2 | 10% |
| 4 | 20% |
| 8 | 22% |
| 16 | 38% | 

At the very least, having them in the benchmark gives us "documentation" of what the expected values are, so we use it in the compiler once it starts doing that automatically.

Reply:
2,8 is for the 16 thread case and the sizes (256x1024). When we do a 32x32 blocking, we get a grid of 8x32 32x32 blocks for the 256x1024 result matrix. 2x8 as parallel task grid means we stamp out 2x8 blocks (and assign them to 1 cores) from the 8x32 task grid. That means the original matrix is now optimal computed in a 4x4 grid (minimal communication).

Having said this, we have to try for all sizes and thread counts which is "good", e.g. I just did this for 8 threads and got best performance for 4,8 , for 4 threads it's 8,8 and for 2 threads it's 8,16

... which is also the chosen libxsmm-dnn config :-) and it makes sense as we minimize weight matrix movement with these settings.

Perhaps we want to change the parameter to #workers in each dimensions instead of tiles/worker in each dimension... However, this we can also fix in a future PR.

Reply:
> Having said this, we have to try for all sizes and thread counts which is "good", e.g. I just did this for 8 threads and got best performance for 4,8 , for 4 threads it's 8,8 and for 2 threads it's 8,16

Good, then for now we just change the benchmarks JSON config to have those options to the pass, so that we "document" that these are the optimal configs for each number of threads.

> Perhaps we want to change the parameter to #workers in each dimensions instead of tiles/worker in each dimension... However, this we can also fix in a future PR.

Indeed, for a future PR. But having the options in the benchmarks JSON files will help whoever tackles this to get there faster, as well as show our best performance for each benchmark.